[12.0][WIP][REF] refactor code, rename enabled to active, and simplifie run code

sebastienbeau opened the pull request:
- [x] rename enabled to active
- [x] unify run button
- [x] improve UX

sebastienbeau committed:
[REF] refactor code, rename enabled to active, and simplifie run code

GSLabIt commented on attachment_synchronize/models/attachment_synchronize_task.py:
```diff
@@ -89,7 +89,7 @@ class AttachmentSynchronizeTask(models.Model):
         "on their 'File Type' value.",
     )
-    enabled = fields.Boolean("Enabled", default=True)
+    active = fields.Boolean("Enabled", default=True, old="enabled")
```
if task is deactivated will disappear from list, that's why used enabled instead of active. is this a wanted feature?

bealdav reviewed (commented):
Why remove enable?

sebastienbeau commented:
@bealdav  @GSLabIt 

The aim is to stick to odoo logic (toggle active is here natively).

Regarding the visibility it's really easy to solve it, you just need to add an "{active_test: False} in the context
Like here : https://github.com/OCA/server-tools/pull/1917/files#diff-043c362866d526d479476b871011f828b7df1faa5b2daba6b59d96088f65a925R121

So we have exactly the same UI and the same behaviour, just less code and more odoo like

GSLabIt commented:
> @bealdav @GSLabIt
> 
> The aim is to stick to odoo logic (toggle active is here natively).
> 
> Regarding the visibility it's really easy to solve it, you just need to add an "{active_test: False} in the context
> Like here : https://github.com/OCA/server-tools/pull/1917/files#diff-043c362866d526d479476b871011f828b7df1faa5b2daba6b59d96088f65a925R121
> 
> So we have exactly the same UI and the same behaviour, just less code and more odoo like

Yup, you're right! I've seen it after writing the comment; so i deleted it.

sebastienbeau committed:
[IMP] add smart boutton on form view and action button with metric on tree view

bealdav commented:
I know active test but I think it's a bad implementation because it's rely on context. Sometimes you come into a view from a secondary path without any context and then your view is broken.Real example. For me we have to go away from active test.

GSLabIt commented:
> I know active test but I think it's a bad implementation because it's rely on context. Sometimes you come into a view from a secondary path without any context and then your view is broken.Real example. For me we have to go away from active test.

if you call action, it will add context; if you create custom one need to remember add `active_test=False` to the context. Imho, better keep tight to odoo standards.

sebastienbeau committed:
[REF] make menu more consistent

In normal tree view duplicate and archive can be done throught the
action menu. So we only need to show this button when this tree view
are included in a form using a o2m or m2m.
So this visibility can be activated by passing value in the context

Also make the copy consitent whatever way you do it

sebastienbeau commented:
I have added some action directly accessible from the tree view
And duplicate/archive button are hidden by default

![image](https://user-images.githubusercontent.com/1164578/97051708-ef9ce980-157f-11eb-9b10-3534762d36b7.png)


You can still show active/duplicate button through the context. Like in my case for pattern_import_export_synchronize when I show them using a one2many

![image](https://user-images.githubusercontent.com/1164578/97052275-faa44980-1580-11eb-9268-a6b59ca4f152.png)

sebastienbeau commented:
@bealdav the issue with enabled is that
- you duplicate the code
- you have to add domain in several place if you when to hide the record not enable
In this case we only when to show this task in some specific place, and I prefers by default to hide disable task.

In any case I had to add the active_test only in two place (one here in the action, then one time in pattern_import_export_synchronize). I think it's cleaner, let's see what @florian-dacosta think

bealdav commented:
@sebastienbeau ok it you're convince of that.

florian-dacosta commented:
Hi @bealdav @sebastienbeau 
I also prefer to go back to active to be consistent with standard Odoo.

For me, it is not even important to pass the context active test, as I don't really see the point of seeing by default unactive records, but I don't mind if it is the case.
Anyway, even if the context is not passed for a reason or another, it is no big deal, you rarely have to deal with  inactive stuff.

bealdav reviewed (approved):
FYI, this module and its dependency needs to be hardly refactored from 13.0 because of this
https://github.com/OCA/edi/pull/243#discussion_r509216851
Don't waste too many time to fine tuning it for now. Go with that.

sebastienbeau commented:
/ocabot merge major
